Mathlib/Topology/Algebra/Category/ProfiniteGrp.lean

```diff
@@ -161,4 +165,169 @@ instance : HasForget₂ ProfiniteGrp Grp where
     map := fun f => f.toMonoidHom
   }
 
+/-- A topological group that has a ContinuousMulEquiv to a profinite group is profinite -/
+def ofContinuousMulEquivProfiniteGrp {G : ProfiniteGrp.{u}} {H : Type v} [TopologicalSpace H]
+    [Group H] [TopologicalGroup H] (e : ContinuousMulEquiv G H) : ProfiniteGrp.{v} :=
+  letI : CompactSpace H := Homeomorph.compactSpace e.toHomeomorph
+  letI : TotallyDisconnectedSpace G := Profinite.instTotallyDisconnectedSpaceαTopologicalSpaceToTop
+  letI : TotallyDisconnectedSpace H := Homeomorph.totallyDisconnectedSpace e.toHomeomorph
+  .of H
+
+/-- The closed subgroup of a profinite group is profinite -/
+def ofClosedSubgroup {G : ProfiniteGrp}
+    (H : Subgroup G) (hH : IsClosed (H : Set G)) : ProfiniteGrp :=
+  letI : CompactSpace H := isCompact_iff_compactSpace.mp (IsClosed.isCompact hH)
+  of H
+
+end ProfiniteGrp
+
+/-!
+# The projective limit of finite groups is profinite
+
+* `FiniteGrp.limit` : the concretely constructed limit of finite group as a subgroup of the pi-type
+
+* `ofFiniteGrpLimit`: direct limit of finite groups is a profinite group
+
+* Verify that the constructed limit satisfies the universal property.
+-/
+
+section Profiniteoflimit
+
+open TopologicalGroup
+
+universe w w'
+
+variable {J : Type v} [Category.{w, v} J] (F : J ⥤ FiniteGrp.{max v w'})
```

Comment:
Will this cause problems if I am working in `FiniteGrp.{u}` and I have some category `J` which is just Nat with a -> b if a >= b? Then w=v=0.

Reply:
I think it is jujian who design the universe level, sorry that I don't know clearly about these problems.

Reply:
@jjaassoonn This is for you then 😄

Reply:
No, the current design asks the universe level of the the finite groups to be bigger than that of the indexing category:

```lean
#check @limit ℕ _
-- limit : (F : ℕ ⥤ ProfiniteGrp.{u_1}) → Subgroup ((j : ℕ) → ↑(F.obj j).toProfinite.toTop)
```

So if the indexing category, the `ProfiniteGrp` can be anything